### PR TITLE
6160 data backed fragments

### PR DESCRIPTION
**Pull request by ttoomey**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Dependencies

This branch has the following dependencies. These must be merged before this branch:
* https://github.com/greenriver/hmis-warehouse/pull/4422
* https://github.com/open-path/Green-River/issues/6164
* https://github.com/greenriver/hmis-warehouse/pull/4484

## Description

[GH Issue](https://github.com/open-path/Green-River/issues/6160)

Migrates from file-backed fragments to db fragments (DefinitionFragment). Adjust seed script to populate the definitions from file-backed fragments. This enables full customization of client forms and allows for future client-managed fragments (or fragment library) feature. Also added fingerprinting function for testing to more easily compare how code changes impact FDs

After this is deployed:
* default definition fragments will still be populated from fragment files
* fragments that are customized via env-specific fragments or patches are no longer seeded on deploy (after they are initially created)
* env-specific fragments and  patches can be removed _pending resolution of question below_

Questions:
* env-specific fragments apply changes to form definitions directly, not just fragments. If the patches are removed, the form definitions will be seeded with default values, losing the env-specific customizations. Do we need a to track "system_managed" form definitions (similar to how we track this for fragments in this pr)? This would allow us to flag customized FDs and stop seeding them

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis_external_apis/public/schemas/form_definition.json
Comment (ttoomey) [inline drivers/hmis_external_apis/public/schemas/form_definition.json:229]: Thought it would be better to use a different property name than 'fragment' since it has a different meaning and format than the current 'fragment' property

**Review by gigxz — APPROVED** [review]
looks good!

**Comment by ttoomey** [comment]
tbd: adjustments to initial fragment seeding
- [x] apply patches to fragments
- ~~resolve nested fragments (I think this is needed to apply patches reliably)~~

**Comment by ttoomey** [comment]
should be good to merge in after the dependent branches are merged

**Pull request closed without merging**
